### DISCUSSION TITLE
【新提案】按字节比较 ArrayBuffer 相等

### DISCUSSION THREAD

**Discussion by septs** [JSCIG/es-discuss#15] [Ideas]
```typescript
ArrayBuffer.isEquals(a, b)
```

https://github.com/NiceLabs/proposal-arraybuffer-equals
https://es.discourse.group/t/compare-two-arraybuffers-bit-wise-for-equality/413

**Comment by jjeejj** [comment]
这个不就是 导致 空的长度相等的 buffer 都是相等的（在申请buffer时，默认填充0的）

**Comment by septs** [comment]
@jjeejj 逻辑上并没有问题
这个函数的关注焦点应该是比较两个 ArrayBuffer 在逻辑上是否完全相等
即使这个两个 ArrayBuffer 都是 all zero

两个认为两个长度为 0 的 ArrayBuffer 应当不等，那么是逻辑错误
（对于使用者来说关心正是 a 在逻辑上完全等效于 b）

**Comment by magic-akari** [comment]
这是不是比较值相等呢？是否也是 https://github.com/tc39/proposal-record-tuple 要解决的问题呢？

**Comment by septs** [comment]
@magic-akari 根本性问题是，不能改变既有的语义
break the web (world)

**Comment by Jack-Works** [comment]
> 这是不是比较值相等呢？是否也是 https://github.com/tc39/proposal-record-tuple 要解决的问题呢？

https://github.com/tc39/proposal-record-tuple/issues/134

**Comment by septs** [comment]
#18
